src/compiler.c

```diff
@@ -5254,8 +5254,8 @@ Int CompileFunc (
     compilerMagic2 = magic2;
 
     /* create 'CompInfoGVar' and 'CompInfoRNam'                            */
-    CompInfoGVar = NewBag( T_STRING, sizeof(UInt) * 1024 );
-    CompInfoRNam = NewBag( T_STRING, sizeof(UInt) * 1024 );
+    CompInfoGVar = NewBag( T_DATOBJ, sizeof(UInt) * 1024 );
+    CompInfoRNam = NewBag( T_DATOBJ, sizeof(UInt) * 1024 );
```

Comment:
This part of the change conflicts with PR #3527 in a trivial way.

Reply:
I also find the end of the commit message a bit confusing. It says:
> The values stored (RNams and GVars) are never 0, so there are no
problems with overwriting the 0th element (where the type of a
DATOBJ is stored).

I think this only refers to `CompInfoGVar` and `CompInfoRNam`, not `info`, but that's IMHO not so clear if one looks at this commit w/o prior knowledge, and then is mildly confusing. But even then it is IMHO confusing, because we do not store the RNam and GVar; but rather, we use them, treated as UInt, and then use them as an offset into `(UInt *)ADDR_OBJ(CompInfoGVar)` (resp. RNam).. And these offsets always happen to be at least 1, so we never overwrite slot 0.

Perhaps just drop these two lines from the commit, and that paragraph from the PR; then our two PRs don't conflict anymore?